### DISCUSSION TITLE
Z2M : migrate from SONOFF ZBDongle-P to SONOFF ZBDongle-E for 70+ devices paired

### DISCUSSION THREAD

**Discussion by N00zy** [Koenkk/zigbee2mqtt#22992] [Devices]
Hi,
I'm using Z2M in Home Assistant (rpi4, HAOS) with a SONOFF ZBDongle-P on which 70+ devices are paired. Everything works fine so far.
I recently purchased a SONOFF ZBDongle-E on which I installed the latest Ember firmware (v7.4.2.0).
I would like to migrate from the P dongle to the E one (for better radio, Thread support...), if possible without having to repair all devices, as it took me a lot of time to set up the current config so far :-)
The reason why I did not use the E dongle one year ago is because, at that time, it was still in experimental support for Z2M, but now it seems that it is fully supported by Z2M. Can someone please confirm this?
Concerning the migration from the P to the E dongle, any help for easy processing would be highly appreciated.
Thanks a lot for your help!
Kind regards,
Greg

**Comment by Koenkk** [comment]
Tbh, if the dongle P works fine I would not migrate. Where did you get the info that the radio is better? Also, the multiprotocol fw is not recommended and stable so you will not get Thread support anyway.
